src/components/Form/formFunctions.ts

```diff
@@ -0,0 +1,181 @@
+import React from "react";
+import { booleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import { Database } from "@/database_types_file";
+import {
+    changeEventHandler,
+    selectChangeEventHandler,
+} from "@/components/DataInput/inputHandlerFactories";
+
+type Field = Fields[keyof Fields];
+export type ErrorSetter = (errorKey: string, errorType: Error) => void;
+export type FieldSetter = (fieldKey: string, newFieldValue: Field) => void;
+export type PersonType = Database["public"]["Enums"]["gender"];
+
+export enum Error {
+    initial = "required",
+    none = "",
+    required = "This is a required field.",
+    invalid = "Please enter a valid entry.",
+    submit = "Please ensure all fields have been entered correctly. Required fields are labelled with an asterisk.",
+    database = "An error has occurred. Please try again later.",
+}
+
+export const numberRegex = /^\d+$/;
+
+export interface CardProps {
+    formErrors: FormErrors;
+    errorSetter: ErrorSetter;
+    fieldSetter: FieldSetter;
+    fields: Fields;
+}
+
+export interface Person {
+    personType: PersonType;
+    age?: number | null;
+    quantity?: number;
+}
+
+export interface FormErrors {
+    fullName: Error;
+    phoneNumber: Error;
+    addressLine1: Error;
+    addressPostcode: Error;
+    adults: Error;
+    numberChildren: Error;
+    nappySize: Error;
+}
+
+export interface Fields {
+    fullName: string;
+    phoneNumber: string;
+    addressLine1: string;
+    addressLine2: string;
+    addressTown: string;
+    addressCounty: string;
+    addressPostcode: string;
+    adults: Person[];
+    numberChildren: number;
+    children: Person[];
+    dietaryRequirements: booleanGroup;
+    feminineProducts: booleanGroup;
+    babyProducts: boolean | null;
+    nappySize: string;
+    petFood: booleanGroup;
+    otherItems: booleanGroup;
+    deliveryInstructions: string;
+    extraInformation: string;
+    attentionFlag: boolean;
+    signpostingCall: boolean;
+}
```

Comment:
I don't think these interfaces should be in the general `Form/formFunctions.ts` file as they are rather specific to the `AddClientsForm`. I'd perhaps move them into the `AddClientsForm.tsx` file and make general `Field` and `Error` interfaces in this folder. You can then say that the interfaces defined in the `AddClientsForm.tsx` file `extend` the more general interfaces defined in `formFunctions.ts`